scripts/environment.sh

```diff
@@ -86,23 +86,29 @@ case "${GINKGO_FOCUS:-}" in
     export WORKER_MACHINE_COUNT=${WORKER_MACHINE_COUNT:-"0"}
     export KUBERNETES_VERSION_UPGRADE_FROM=${FROM_K8S_VERSION}
   ;;
-esac
 
-# IPReuse feature test environment vars and config
-if [[ ${GINKGO_FOCUS:-} == "features" && ${GINKGO_SKIP:-} == "pivoting remediation" ]]; then
-  export NUM_NODES="5"
-  export CONTROL_PLANE_MACHINE_COUNT=${CONTROL_PLANE_MACHINE_COUNT:-"3"}
-  export WORKER_MACHINE_COUNT=${WORKER_MACHINE_COUNT:-"2"}
-fi
+  # K8s conformance test environment vars and config
+  # The test needs 6 nodes in total because the tests run in parallel on 5 worker machines:
+  # https://github.com/kubernetes-sigs/cluster-api/blob/main/test/e2e/k8s_conformance.go#L104
+  k8s-conformance)
+    export NUM_NODES="6"
+    export CONTROL_PLANE_MACHINE_COUNT=${CONTROL_PLANE_MACHINE_COUNT:-"1"}
+    export WORKER_MACHINE_COUNT=${WORKER_MACHINE_COUNT:-"5"}
+  ;;
 
-# K8s conformance test environment vars and config
-# The test needs 6 nodes in total because the tests run in parallel on 5 worker machines:
-# https://github.com/kubernetes-sigs/cluster-api/blob/main/test/e2e/k8s_conformance.go#L104
-if [[ ${GINKGO_FOCUS:-} == "k8s-conformance" ]]; then
-  export NUM_NODES="6"
-  export CONTROL_PLANE_MACHINE_COUNT=${CONTROL_PLANE_MACHINE_COUNT:-"1"}
-  export WORKER_MACHINE_COUNT=${WORKER_MACHINE_COUNT:-"5"}
-fi
+  # IPReuse feature test environment vars and config
+  features)
```

Comment:
I was confused for a minute that how come this is called feature while we have the different features in here https://github.com/metal3-io/cluster-api-provider-metal3/pull/2607/files#diff-f3736fd0d9f6c379e7622c6230d742bb019c1120223e1c3857bb1808d27d5413L75
But now I see this is ipreuse test. so how come this needs a bigger env if the pivot and remediation is skipped?

Reply:
I just kept the existing numbers there and used the numbers @adilGhaffarDev said are correct for this. Note that these are also fallback variables, so they might be overridden somewhere else like JJB. Based on the remediation job output, these 3+1 numbers are correctly used now though.

Reply:
yeah all makes sense, I am just confused by the 5 node requirement for any case  but Adil might know the reason.

Reply:
5 nodes are only for IP-reuse and clusterctl upgrade tests. 
Here is where we changed it for ip-reuse: https://github.com/metal3-io/cluster-api-provider-metal3/pull/2201 
For clusterctl, we are using the CAPI test, which I believe requires 5 nodes.

For IP-reuse we can improve it, maybe try to create a similar scenario with fewer nodes where we can make sure IP-reuse is working. But I need to check that. I remember there were some issues.